demo/vue3js_dynamic-render_grid-item.html

```diff
@@ -0,0 +1,168 @@
+      import { createApp, ref, onMounted, onBeforeUnmount, h, render, toRefs } from "https://cdn.jsdelivr.net/npm/vue@3.0.11/dist/vue.esm-browser.js";
+
+      const GridItemComponent = {
+        props: {
+          itemId: {
+            type: [String, Number],
+            required: true,
+          },
+        },
+        emits: ['remove'],
+        setup(props, { emit }) {
+          const root = ref(null)
+          const { itemId } = toRefs(props)
+
+          onBeforeUnmount(() => {
+            console.log(`In vue onBeforeUnmount for item ${itemId.value}`)
+          });
+
+          function handleRemove() {
+            emit('remove', root.value)
+          }
+
+          return {
+            root,
+            itemId,
+            handleRemove,
+          }
+        },
+        template: `
+          <div ref="root" class="grid-stack-item my-custom-grid-item-component">
+            <div class="grid-stack-item-content">
+              <button @click=handleRemove>X</button>
+              <p>
+                Vue Grid Item {{ itemId }}
+              </p>
+            </div>
+          </div>
+        `
+      }
+
+      createApp({
+        setup() {
+          let info = ref("");
+          let grid = null; // DO NOT use ref(null) as proxies GS will break all logic when comparing structures... see https://github.com/gridstack/gridstack.js/issues/2115
+          const items = [
+            { id: 1, x: 2, y: 1, h: 2 },
+            { id: 2, x: 2, y: 4, w: 3 },
+            { id: 3, x: 4, y: 2 },
+            { id: 4, x: 3, y: 1, h: 2 },
+            { id: 5, x: 0, y: 6, w: 2, h: 2 },
+          ];
+          let count = ref(items.length);
+          const shadowDom = {}
+
+          onMounted(() => {
+            grid = GridStack.init({ // DO NOT user grid.value = GridStack.init(), see above
+              float: true,
+              cellHeight: "70px",
+              minRow: 1,
+            });
+
+            GridStack.addRemoveCB = gsAddRemoveVueComponents;
+
+            grid.load(items);
+          });
+
+          function gsAddRemoveVueComponents(host, item, add, isGrid) {
+            if (!host) {
+              return
+            }
+
+            // Not supported yet
+            if (isGrid) {
+              return;
+            }
+
+            if (add) {
+              const itemId = item.id
+
+              // Use Vue's render function to create the content
+              // See https://vuejs.org/guide/extras/render-function.html#render-functions-jsx
+              //      Supports: emit, slots, props, attrs, see onRemove event below
+              const itemVNode = h(
+                GridItemComponent,
+                {
+                  itemId: itemId,
+                  onRemove: (itemEl) => {
+                    grid.removeWidget(itemEl)
+                  }
+                }
+              )
+              shadowDom[itemId] = document.createElement('div')
+              render(itemVNode, shadowDom[itemId])
```

Comment:
These two lines are the most important aspect of this code, and the area I'm the most unsure of. 

I haven't used Vue's render functions before, and I wasn't able to figure out how to dynamically render the component w/o creating this "shadow" element.